packages/core/src/client.ts

```diff
@@ -240,132 +252,85 @@ export class SkipRouter {
-          onTransactionTracked: options.onTransactionTracked,
-        });
 
-        if (options.onTransactionCompleted) {
-          await options.onTransactionCompleted(
-            multiChainMsg.chainID,
-            tx.transactionHash,
-            txStatusResponse,
-          );
+        txResult = {
+          chainID: multiChainMessage.chainID,
+          txHash: txResponse.transactionHash
+        }
+      } else if ("evmTx" in message) {
+        const txResponse = await this.executeEvmMultiChainMsg(message, options);
+        txResult = {
+          chainID: message.evmTx.chainID,
+          txHash: txResponse.transactionHash,
         }
+      } else {
+        raise(`executeRoute error: invalid message type`);
       }
 
-      if ("evmTx" in message) {
-        const { evmTx } = message;
+      if (options.onTransactionBroadcast) {
+        await options.onTransactionBroadcast({ ...txResult });
+      }
 
-        const getEVMSigner = options.getEVMSigner || this.getEVMSigner;
-        if (!getEVMSigner) {
-          throw new Error(
-            "executeRoute error: 'getEVMSigner' is not provided or configured in skip router",
-          );
-        }
+      const txStatusResponse = await this.waitForTransaction({
+        ...txResult,
+        onTransactionTracked: options.onTransactionTracked,
+      });
 
-        const evmSigner = await getEVMSigner(evmTx.chainID);
+      if (options.onTransactionCompleted) {
+        await options.onTransactionCompleted(
+          txResult.chainID,
+          txResult.txHash,
+          txStatusResponse,
+        );
+      }
+    }
+  }
 
-        const txReceipt = await this.executeEVMTransaction({
-          message: evmTx,
-          signer: evmSigner,
-        });
+  private async executeEvmMultiChainMsg(message: { evmTx: types.EvmTx; }, options: clientTypes.ExecuteRouteOptions) {
+    const { evmTx } = message;
 
-        if (options.onTransactionBroadcast) {
-          await options.onTransactionBroadcast({
-            chainID: evmTx.chainID,
-            txHash: txReceipt.transactionHash,
-          });
-        }
+    let getEVMSigner = options.getEVMSigner || this.getEVMSigner;
+    if (!getEVMSigner) {
+      throw new Error("Unable to get EVM signer");
+    }
 
-        const txStatusResponse = await this.waitForTransaction({
-          chainID: evmTx.chainID,
-          txHash: txReceipt.transactionHash,
-          onTransactionTracked: options.onTransactionTracked,
-        });
+    const evmSigner = await getEVMSigner(evmTx.chainID);
 
-        if (options.onTransactionCompleted) {
-          await options.onTransactionCompleted(
-            evmTx.chainID,
-            txReceipt.transactionHash,
-            txStatusResponse,
-          );
-        }
-      }
-    }
+    return await this.executeEVMTransaction({
+      message: evmTx,
+      signer: evmSigner,
+    });
   }
 
-  async executeMultiChainMessage(
+  async executeCosmosMultiChainMessage(
```

Comment:
Can we instead name this `executeCosmosMessage`.

And in addition can we leave `executeMultiChainMessage`, which wraps a new `executeCosmosMessage` method, and mark it as deprecated.

Reply:
did this in the [latest commit](https://github.com/skip-mev/skip-router-sdk/pull/116/commits/045096399cc7c99947879de6e43c03c8942270f5)